sympy/integrals/tests/test_prde.py

```diff
@@ -144,6 +146,41 @@ def test_prde_no_cancel():
     assert (Matrix([q])*V[0][:6, :])[0] == Poly(x - 1/2, t, domain='QQ(x)')
 
 
+def test_prde_cancel_liouvillian():
+    ### 1. case == 'primitive'
+    # used when integrating (log(-1/y)/2 - log(1/y)/2)/y - (log(1 - 1/y)/2 - log(1 + 1/y)/2)/y
+    # Not taken from 'the' book
+    DE = DifferentialExtension(extension={'D': [Poly(1, x), Poly(1/x, t)]})
+    assert prde_cancel_liouvillian(Poly(-1/(x - 1), t), [Poly(-x + 1, t), Poly(1, t)], 1, DE) == \
```

Comment:
It looks OK. Apparently the highest level with `D` given by `Poly(1/(x - 1), _t1, domain='ZZ(x)')` is not needed in this application of `prde_cancel_liouvillian`.